### PR TITLE
Feat: Update Transcription service

### PR DESCRIPTION
**Pull request by jeremyalv**
# Overview
This pull request implements the `Storing Transcription` feature to the app. Previously, for sprint 1 we used the just-in-time generated Transcription data to be displayed on the `View Transcription` screen. For sprint 2, we will introduce persistent storage for Transcription data so that we can implement the `My Transcriptions` feature and users can go back to any Transcriptions they create previously.

# Major Features
* Store `Transcription` and `TranscriptionChunk` objects to the database when users call the create transcription endpoint
* Create a unified Migrations Script inside `alembic/` folder
* Added & Updated Model and Schema for Transcription Service 
* Remigrated Dev + Prod Database 

# Minor Features
* Refactored View Transcription backend service (for Separation of Concerns)
* Updated Controller Response types to use `JSONResponse`
* Return JSON-serialized Pydantic objects for all `Transcription` controller return values
* Refactored bash scripts to `scripts/` and add script for swapping PROD/DEV environment variables
* Add `waitlist` table to the database for Landing Page
* Refactored by removing unused imports, deleting comments, and more
* Fixes by correcting UUID imports, `setting some Migration tables to use nullable=True`, and more

### CODE REVIEW

File: src/services/email_verification.py [hunk 1]
```diff
@@ -39,149 +40,148 @@ def is_user_exists(session: Session, payload: CheckUserExistsSchema) -> bool:
...
+            recipients=[recipient],
+            body=MESSAGE_BODY,
+            subtype=MessageType.html,
+         )
+
+         client = get_mail_client()
+
+         await client.send_message(message)
+
+         return GenericResponseModel(
```
Comment (daffafaizan) [inline src/services/email_verification.py:166]: After reviewing your changes in other files, I believe this should also be changed to ```JSONResponse```

File: src/services/transcription.py [hunk 2]
```diff
@@ -80,17 +97,178 @@ async def transcribe_file(
...
+
+            return db_tsc_chunk
+        except Exception as e:
+            session.rollback()
+            raise RuntimeError(f"Error while inserting Transcription Chunk to DB: {e}")
+        finally:
+            session.close()
+
+    async def retrieve_formatted_transcription_from_job_name(
+        self,
```
Comment (amrul-hzz) [inline src/services/transcription.py:151]: `retrieve_formatted_transcription_from_job_name()` can be broken down into its more atomic subprocesses. For example:

```
 async def retrieve_formatted_transcription_from_job_name(
        self, transcribe_client, job_name: str
    ):
        transcription_data = await self._fetch_transcription_data(transcribe_client, job_name)
        processed_data = self._process_transcription_data(transcription_data)
        response = self._format_transcription_response(processed_data)
        return response

    async def _fetch_transcription_data(self, transcribe_client, job_name: str):
        response = await self.get_all_transcriptions(transcribe_client=transcribe_client, job_name=job_name)
        aws_link = requests.get(response)
        return aws_link.json()

    def _process_transcription_data(self, transcription_data):
        job_name = transcription_data.get("jobName")
        account_id = transcription_data.get("accountId")
        status = transcription_data.get("status")
        transcripts = transcription_data.get("results", {}).get("transcripts", [])
        full_transcript = " ".join([t.get("transcript") for t in transcripts if t.get("transcript")])
        items = transcription_data.get("results", {}).get("items", [])
        return job_name, account_id, status, full_transcript, items

    def _format_transcription_response(self, processed_data):
        job_name, account_id, status, full_transcript, items = processed_data
        grouped_items = self._generate_grouped_items_and_format_chunks(items=items)
        return {
            "jobName": job_name,
            "accountId": account_id,
            "status": status,
            "results": {
                "transcripts": full_transcript, 
                "items": grouped_items
            },
        }
```
I think writing it this way makes it easier to follow, especially when the function names can be quite descriptive of what is done to the data after being fetched before returned as a response.

**Review by amrul-hzz — APPROVED** [review]

**Comment by annavaws** [comment]
Consider replacing multiple print statements with logging for better error handling and consistency in logging practices. You can refer to this logging guide https://www.machinelearningplus.com/python/python-logging-guide/

File: src/services/transcription.py (hunk 2 above)
Comment (jeremyalv) [reply src/services/transcription.py:151]: Thanks for the suggestion. I've kept some parts that are deemed fit for the method. Won't include all bcs it introduces new bugs - let's refactor when we have more time in week 2

**Comment by jeremyalv** [comment]
> Consider replacing multiple print statements with logging for better error handling and consistency in logging practices. You can refer to this logging guide https://www.machinelearningplus.com/python/python-logging-guide/

Noted, will consider adding logging statements within the code

**Review by daffafaizan — APPROVED** [review]

File: src/services/email_verification.py (hunk 1 above)
Comment (jeremyalv) [reply src/services/email_verification.py]: thanks for the suggestion, implemented it

**Pull request merged**